portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,56 +34,40 @@
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
-
-  //private ArrayList<String> quotes;
-  ArrayList<String> comment_text = new ArrayList<String>();
-  ArrayList<String> author_name = new ArrayList<String>();
 
+public class DataServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-   
-    List<String> quotes = new ArrayList<String>();
-    String quoteOne = "I am in a charming state of confusion. - Ada Lovelace";
-    String quoteTwo = "It is much easier to apologise than it is to get permission. - Grace Hopper";
-    String quoteThree = "You can do anything you want to, but you have to work at it - Annie Easley";
-    quotes.add(quoteOne);
-    quotes.add(quoteTwo);
-    quotes.add(quoteThree);
-    
-    
-    String json = convertToJsonUsingGson(quotes);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Message> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String text = (String) entity.getProperty("text");
+      String comment_author = (String) entity.getProperty("comment_author");
+      long timestamp = (long) entity.getProperty("timestamp");
+      long id = entity.getKey().getId();
+
+      Message post = new Message(text, comment_author, timestamp, id);
+      comments.add(post);
+    }
+    Gson gson = new Gson();
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(gson.toJson(comments));
 
   }
-
   /**
-   * Converts a ServerStats instance into a JSON string using GSON
+   * Converts a  instance into a JSON string using GSON
    */
   private String convertToJsonUsingGson(List<String> quotesToGson) {
     Gson gson = new Gson();
     String json = gson.toJson(quotesToGson);
     return json;
   }
-
-public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the input from the form.
-    String text = getParameter(request, "text-input", "");
-    String comment_author = getParameter(request, "comment_author", "");
-    comment_text.add(text);
-    author_name.add(comment_author);
-    // Respond with the result.
-    response.setContentType("text/html;");
-    response.getWriter().println(author_name.get(0) + ": "+ comment_text.get(0));
-    // Redirect back to the HTML page.
-    response.sendRedirect("/index.html");
-  }
-
   /**
    * @return the request parameter, or the default value if the parameter
-   *         was not specified by the client*/
-   
+   *         was not specified by the client*/  
   private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
Delete this method if you no longer use it in this class?

Reply:
I use .getParameter in my deleteservlet.java and newmessage.java, is that a totally different method or is it somehow calling upon this ?